src/marqo/core/document/document.py

```diff
@@ -13,12 +19,84 @@ def __init__(self, vespa_client: VespaClient, index_management: IndexManagement)
         self.vespa_client = vespa_client
         self.index_management = index_management
 
-    def delete_all_docs(self, index_name: str) -> int:
-        """Deletes all documents in the given index"""
+    def delete_all_docs_by_index_name(self, index_name: str) -> int:
+        """Deletes all documents in the given index by index name"""
         if not self.index_management.index_exists(index_name):
             raise IndexNotFoundError(f"Index {index_name} does not exist")
 
         marqo_index = self.index_management.get_index(index_name)
-        res: DeleteAllDocumentsResponse = self.vespa_client.delete_all_docs(marqo_index.schema_name)
+        return self.delete_all_docs(marqo_index)
 
+    def delete_all_docs(self, marqo_index):
+        """Deletes all documents in the given index by marqo_index object"""
+        res: DeleteAllDocumentsResponse = self.vespa_client.delete_all_docs(marqo_index.schema_name)
         return res.document_count
+
+    def partial_update_documents_by_index_name(self, index_name,
+                                               update_documents_params: UpdateDocumentsBodyParams) -> Dict:
+        if not self.index_management.index_exists(index_name):
+            raise IndexNotFoundError(f"Index {index_name} does not exist")
+
+        marqo_index = self.index_management.get_index(index_name)
+        if marqo_index.type == IndexType.Structured:
+            return self.structured_partial_update_documents(update_documents_params.documents, marqo_index)
+        elif marqo_index.type == IndexType.Unstructured:
+            raise UnsupportedFeatureError("'Partial document update' is not supported for unstructured indexes. "
+                                          "Please use 'add_documents' with 'use_existing_tensor=True' instead.")
+        else:
+            ValueError(f"No known implementation for index type {marqo_index.type}")
+
+    def structured_partial_update_documents(self, documents, marqo_index):
```

Comment:
Ideally we don't want to have an index type distinction here. The Vespa index abstraction should handle this similar to how search is implemented

Reply:
Done.